virtual_labs/tests/utils.py

```diff
@@ -243,3 +247,18 @@ async def create_confirmed_setup_intent(customer_id: str) -> SetupIntent:
     )
 
     return intent
+
+
+async def wait_until(
+    somepredicate: typing.Any,
+    timeout: int,
+    period: float = 0.25,
+    *args: typing.Any,
+    **kwargs: typing.Any,
+) -> bool:
+    mustend = time.time() + timeout
+    while time.time() < mustend:
+        if somepredicate(*args, **kwargs):
```

Comment:
Actually, @Dinika I just realised this. This function isn't working as intended because `somepredicate` blocks. That is the code is stuck in this `if` until `somepredicate` returns even after `timeout` has passed.

To make this work you would need to either:

1. make `somepredicate` async and use `asyncio.wait_for` (https://docs.python.org/3/library/asyncio-task.html#asyncio.wait_for)

2.  Or if `somepredicate` has to by synchronous then run it in a separate thread.

But you don't even need `wait_until` since you can simply make the call with `httpx`.  I'll comment above.

Reply:
Ah, I realised the goal is not to time out on `somepredicate` but on the nexus views to be created.  Good, then this works, however it would still be beneficial to make the `somepredicate`  async. and await it.

So:

```python
async def wait_until(
    somepredicate: typing.Callable[..., Awaitable[bool]],
    timeout: int,
    period: float = 0.25,
    *args: typing.Any,
    **kwargs: typing.Any,
) -> bool:
    mustend = time.time() + timeout
    while time.time() < mustend:
        if await somepredicate(*args, **kwargs):
            return True
        await asyncio.sleep(period)
    return False
```